libraries/esp8266/examples/DelegatePerf/TestPrep.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+  ESP8266 Delegate class template test by Dirk O. Kaar
+  This example code is in the public domain
+*/
+
+#include "Foo.h"
+#ifndef NODELEGATE
+#include <Delegate.h>
+#else
+#include <functional>
+#endif
+#include <memory>
+
+extern void stopwatch();
+
+std::shared_ptr<Foo> oPtr(new Foo());
+bool inline cbCPtr(int result) {
+  return oPtr->cb(result);
+}
```

Comment:
I have not yet looked at this PR in depth, but one potential issue (if these calls are only run one time) could be the I-cache.

Make all of the routines `ICACHE_RAM_ATTR` and rerunning would be a way to ensure the cache misses don't affect the times (because loading the I$ from SPI flash is painfully slow).